### PR TITLE
Implement fetch metadata for K8s

### LINKED ISSUE #1873: Add MetadataPolicy=Fetch for Kubernetes
`MetadataPolicy: Fetch` is already implemented for Azure KeyVault. We need to implement it for Kubernetes provider in order to allow information to be obtained from `secret.metadata.annotations` and `secret.metadata.labels`. Labels and Annotations should be obtained from the `remoteRef`, by selecting a property like:

```
data:
- metadataPolicy: Fetch
   secretKey: my-key
   remoteRef:
     key: labels
     property: "labels.my-own-label"
- metadataPolicy: Fetch
   secretKey: my-key
   remoteRef:
     key: path-to-my-secret
     property: "annotations.my-own-annotation"
```

### PR DESCRIPTION
**Pull request by sebagomez**
Fixes #1873 

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by sebagomez** [comment]
`make reviewable` ends with errors on my environment. anything I should update on my side?

```
  - spec.md
  - guides/disable-cluster-features.md
  - provider/aws-pushsecret.md
  - snippets/provider-aws-access.md
INFO     -  Documentation built in 14.37 seconds
error: error getting config 'user.name'
make[1]: *** [build] Error 1
make: *** [docs] Error 2
```

File: pkg/provider/kubernetes/client.go [hunk 1]
```diff
@@ -69,6 +108,18 @@ func (c *Client) GetSecretMap(ctx context.Context, ref esv1beta1.ExternalSecretD
 	if err != nil {
 		return nil, err
 	}
+	if ref.MetadataPolicy == esv1beta1.ExternalSecretMetadataPolicyFetch {
+		retData := make(map[string][]byte)
```
Comment (moolen) [inline pkg/provider/kubernetes/client.go:112]: `GetSecret()` above supports `ref.Property` which is needed to target either labels or annotations.
I think it would be a huge plus if this would be supported here as well:

i.e. if `ref.Property` is `labels` then all labels are returns as map (same goes for annotations ofc).

**Review by moolen — COMMENTED** [review]
Hey, this looks very good and i especially like that both `labels` and `annotations` are supported, which is a challenge b/c other providers don't have such a thing.

It's awesome to be able to target either `labels` or `annotations` by setting the `property` :+1: This slight detail is missing in `GetSecretMap()` which would be a huge plus for users (see comment).

We need to standardise the way we use the `property` field with `metadataPolicy=Fetch`. The approach you have taken is reasonable and is in line with the use in other providers (treat `property` as gjson expression to extract nested data). I've proposed a [different approach here](https://github.com/external-secrets/external-secrets/issues/1874#issuecomment-1451092848) which i'm going to amend.

Example:
```yaml
data:
  - secretKey: metadata
    remoteRef:
      key: k8s-source-secret
      metadataPolicy: Fetch
      property: labels.example # <--
```

Could you please add an example to the docs so that users are aware of that? It's not obvious how to use this. I was surprised (in a good way!) that this works it would be great to tell users that this is a thing right away :)

File: pkg/provider/kubernetes/client.go (hunk 1 above)
Comment (moolen) [reply pkg/provider/kubernetes/client.go:112]: I guess we don't need to support gjson here b/c labels and annotations are only key/value pairs.

Comment (sebagomez) [reply pkg/provider/kubernetes/client.go:112]: Hey Moritz, let me rephrase this to see if I understood correctly. What you're saying is that if I have a DataFrom as follows
```yaml
   dataFrom:
   - extract:
       key: json-secret
       property: label
```
You would end up with a secret with every label (and the same for annotations). Is that correct?

Comment (sebagomez) [reply pkg/provider/kubernetes/client.go:112]: How do we know we're talking about labels and not values? do we want to support `metadataPolicy: Fetch` also in dataFrom?

Comment (moolen) [reply pkg/provider/kubernetes/client.go:112]: :point_up_2: right, what you've posted plus the `metadataPolicy=Fetch`. I'd also make the `property` value plural: `labels` and `annotations`.
```yaml
   dataFrom:
   - extract:
       key: k8s-source-secret
       metadataPolicy: Fetch
       property: labels
```

Comment (moolen) [reply pkg/provider/kubernetes/client.go]: I tested with this

```yaml
apiVersion: external-secrets.io/v1beta1
kind: ExternalSecret
metadata:
  name: k8s-example
spec:
  refreshInterval: 1h
  secretStoreRef:
    kind: SecretStore
    name: k8s-store
  target:
    name: k8s-example
  dataFrom:
  - extract:
      key: k8s-source-secret
      metadataPolicy: Fetch
      property: labels
```

And expected a secret that contains the label as key/value pairs:

say a secret has 

```yaml
metadata:
  labels:
    example: "baz"
    foo: "bar"
```

I would want to have this:
```yaml
data:
  example: "..."
  foo: "..."
```

Instead i get a json-encoded value for the key `labels`.

```yaml
apiVersion: v1
kind: Secret
data:
  # resembles to: {"example":"baz","foo":"bar"}
  labels: eyJleGFtcGxlIjoiYmF6IiwiZm9vIjoiYmFyIn0=
```

Comment (sebagomez) [reply pkg/provider/kubernetes/client.go]: I thought in this case that was expected... so it's exactly the same from when using it in the `data:`
No problem, I'll fix it

**Review by moolen — APPROVED** [review]
I did another test run, looks very good :rocket: ! Thank you for this :bow:
